src/pages/settings/subscriptions/common/components/Webhook.tsx

```diff
@@ -0,0 +1,144 @@
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2022. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import { Card, Element } from '@invoiceninja/cards';
+import { InputField, SelectField } from '@invoiceninja/forms';
+import { useAccentColor } from 'common/hooks/useAccentColor';
+import { Subscription } from 'common/interfaces/subscription';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { BiPlusCircle } from 'react-icons/bi';
+import { MdClose } from 'react-icons/md';
+import { SubscriptionProps } from './Overview';
+
+export function Webhook(props: SubscriptionProps) {
+  const [t] = useTranslation();
+
+  const accentColor = useAccentColor();
+
+  const { subscription, handleChange, errors } = props;
+
+  const [headerKey, setHeaderKey] = useState<string>('');
+
+  const [headerValue, setHeaderValue] = useState<string>('');
+
+  const headers = Object.entries(
+    subscription.webhook_configuration.post_purchase_headers
+  );
+
+  const handleAddHeader = () => {
+    handleChange(
+      `webhook_configuration.post_purchase_headers.${headerKey}` as keyof Subscription,
+      headerValue
+    );
+
+    setHeaderKey('');
+
+    setHeaderValue('');
+  };
+
+  const handleRemoveHeader = (key: string) => {
+    if (
+      Object.prototype.hasOwnProperty.call(
```

Comment:
Here we check if the header property exists in the `post_purchase_headers` object. You're right maybe a shorter way is to just use the `hasOwn` function, so that has been changed.